fix(server): adjust panorama medata for new image dimensions

meesfrensel opened the pull request:
## Description
Introduced in #23953, the copied metadata contains some pixel values referencing what part of the sphere is covered. These values have to be multplied by the ratio between preview and original size.

https://github.com/immich-app/immich/pull/23953#issuecomment-3568396072

Fixes #24333 

## How Has This Been Tested?
Tested on hte three test panoramas I use: one is (way) bigger than the preview size, one is a little smaller, and one is 'cropped' aka not covering the whole sphere.


## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation if applicable
- [x] I have no unrelated changes in the PR.
- [ ] I have confirmed that any new dependencies are strictly necessary.
- [x] I have written tests for new code (if applicable)
- [x] I have followed naming conventions/patterns in the surrounding code
- [x] All code in `src/services/` uses repositories implementations for database calls, filesystem operations, etc.
- [x] All code in `src/repositories/` is pretty basic/simple and does not have any immich specific logic (that belongs in `src/services/`)

## Please describe to which degree, if any, an LLM was used in creating this pull request.

...

meesfrensel committed:
fix(server): adjust panorama medata for new image dimensions

mertalev commented on server/src/repositories/media.repository.ts:
```diff
@@ -121,19 +121,29 @@ export class MediaRepository {
-          writeArgs: ['-TagsFromFile', source, `-${tagGroup}:all>${tagGroup}:all`, '-overwrite_original'],
-        },
-      );
+      const metadata = await exiftool.readRaw(source, ['-XMP-GPano:all', '-G1', '-s']);
```
Don't make exiftool read the image again here. Use the same data from the exiftool call in metadata extraction.

mertalev reviewed (commented):
A few code style comments, but otherwise it looks right. I do want to get input from @bwees on this because image editing (not landed yet) can throw a serious curve ball here given the preview and full-size images can be cropped and rotated.

mertalev reviewed (commented):
Re: image editing, the consensus is for editing to not support panoramas in the first pass, so the earlier comment about crop/rotation isn't a blocker for this PR.

meesfrensel committed:
chore: get dimensions directly with ExtractResult

meesfrensel committed:
Merge branch 'main' into fix/panorama-metadata

meesfrensel commented:
While re-testing I noticed the broken preview images suddenly worked. Some digging reveiled that an update to 5.4.1 of the viewer merged in #25377 (viewer [changelog](https://github.com/mistic100/Photo-Sphere-Viewer/releases/tag/5.14.1), [commit](https://github.com/mistic100/Photo-Sphere-Viewer/commit/5008ec0587c2a0300b804890883b43fd4a774ca4)) now supports scaling the xmp panodata in the viewer code itself, if `image width != croppedWidth`, by the appropriate amount exactly like I'm doing here.

Technically it would still be correct to apply this PR, but at least the bug is 'fixed' for now.

Before I put the work into addressing your comments, rebasing on main, etc.: do you think this PR still makes sense to finish?
